runtime/src/flash_ctrl_test.rs

```diff
@@ -0,0 +1,215 @@
+// Licensed under the Apache-2.0 license
+
+// Test flash controller driver read, write, erage page
+
+use crate::flash_ctrl;
+use core::cell::Cell;
+use kernel::hil;
+use kernel::hil::flash::{Flash, HasClient};
+use kernel::static_init;
+use kernel::utilities::cells::TakeCell;
+use kernel::{debug, debug_flush_queue};
+
+#[cfg(any(
+    feature = "test-flash-ctrl-erase-page",
+    feature = "test-flash-ctrl-read-write-page"
+))]
+use crate::board::run_kernel_op;
+
+use core::fmt::Write;
+use romtime::println;
+
+pub(crate) fn test_flash_ctrl_init() -> Option<u32> {
+    // Safety: this is run after the board has initialized the chip.
+    let chip = unsafe { crate::CHIP.unwrap() };
+    chip.peripherals.flash_ctrl.init();
+    Some(0)
+}
+
+// Create flash callback struct for testing
+struct FlashCtrlTestCallBack {
+    read_pending: Cell<bool>,
```

Comment:
@swenson, the code seems to be cleaner by grouping the IO state in a struct.  As for the `TakeCell` of IO buffers , it might be better to keep as they're.